streams/src/main/java/org/apache/kafka/streams/kstream/internals/ChangedDeserializer.java

```diff
@@ -39,7 +39,10 @@ public ChangedDeserializer(final Deserializer<T> inner) {
     @Override
     public void setIfUnset(final Deserializer<Void> defaultKeyDeserializer, final Deserializer<T> defaultValueDeserializer) {
```

Comment:
Could we pass the context instead of both deserializers, and simplify to (relying on the context to throw if necessary):
```
if (inner == null) {
  inner = context.valueSerde();
}
```

Reply:
This is what I mentioned in my main comment - these methods are ultimately referenced from `MeteredKeyValueStore`, `MeteredSessionStore`, and `MeteredWindowStore` in methods that have both `StateStoreContexts` and `ProcessorContext`. We can change it so we pass in the context but we'd need two sets of methods, one for state store contexts and one for processor contexts. The reason i didn't go ahead and do that was because it seemed a little messy, just a trade-off I guess. WDYT @mjsax? The other alternative I thought of was making `processorContext` and `StateStoreContext` related so we could only needed one set of functions but that ended up being a decently sized refactor when I tried to do that

Reply:
That's a good point... Could we add an internal interface "SerdeGetter" (for lack of a better name), that both `ProcessroContext` and `StateStoreContext` extends? And `SerdeGetter` has two methods `keySerder()` and `valueSerde()` ? For this case, we don't need any refactoring (I would hope) and could pass `SerdeGetter` here?

Reply:
That's smart, and seems to mostly work. In order to change the actual `setIfUnset` code though, I have to do some casting since the serde in `serdeGetter` is <?> and I'm running into issues with that now... but just for one set of classes 😕. Do you know if there's a better way to handle the generics? @mjsax